File: DRIVER_MANAGEMENT_PROJECT_BACKEND/src/main/java/com/eren/DRIVER_MANAGEMENT_PROJECT_BACKEND/entity/enums/VehicleStatuses.java
package com.eren.DRIVER_MANAGEMENT_PROJECT_BACKEND.entity.enums;

/**
 * Represents the status of a vehicle in the system.
 *
 * <p>This enumeration is used to track the operational condition of vehicles,
 * such as whether they are available for use or undergoing maintenance.</p>
 */
public enum VehicleStatuses {

    /** Vehicle is operational and ready to be assigned to a task. */
    READY_FOR_SERVICE,

    /** Vehicle has a malfunction and is not suitable for service. */
    MALFUNCTION,

    /** Vehicle is damaged and requires repair. */
    DAMAGED,

    /** Vehicle is out of service and not currently in use. */
    OUT_OF_SERVICE,

    /** Vehicle is undergoing maintenance or repair. */
    UNDER_MAINTENANCE
}

File: DRIVERI_MANAGEMENT_PROJECT_BACKEND/Entities/Enums/VehicleStatuses.cs
using System.ComponentModel.DataAnnotations;
namespace Entities.Enums
{
    public enum VehicleStatuses
    {
        [Display(Name ="Ready For Service")]
        READY_FOR_SERVICE = 0,
        [Display(Name = "Malfunction")]
        MALFUNCTION = 1,
        [Display(Name = "Damaged")]
        DAMAGED = 2,
        [Display(Name = "Out of Service")]
        OUT_OF_SERVICE = 3,
        [Display(Name = "Under Maintenance")]
        UNDER_MAINTENANCE = 4,
    }
}
